src/rootfinding.jl

```diff
@@ -0,0 +1,205 @@
+doc"""
+    surfacecrossing(g_old, g_now, eventorder::Int)
+
+Detect if the solution crossed a root of event function `g`. `g_old` represents
+the last-before-current value of event function `g`; `g_now` represents the
+current value of event function `g`; `eventorder` is the order of the derivative
+of the event function `g` whose root we are trying to find. Returns `true` if
+`g_old` and `g_now` have different signs (i.e., if one is positive and the other
+one is negative); otherwise returns `false`.
+"""
+surfacecrossing(g_old::Taylor1{T}, g_now::Taylor1{T}, eventorder::Int) where {T <: Real} = g_old[eventorder]*g_now[eventorder] < zero(T)
+
+surfacecrossing(g_old::Taylor1{Taylor1{T}}, g_now::Taylor1{Taylor1{T}}, eventorder::Int) where {T <: Real} = g_old[eventorder][0]*g_now[eventorder][0] < zero(T)
+
+surfacecrossing(g_old::Taylor1{TaylorN{T}}, g_now::Taylor1{TaylorN{T}}, eventorder::Int) where {T <: Real} = g_old[eventorder][0][1]*g_now[eventorder][0][1] < zero(T)
+
+doc"""
```

Comment:
Why Newton-Rhapson? You have a bounded 1-dimensional interval for the time where the zero is. For large equations its must faster to use an interval-based rootfinding algorithm like bisection or the Illinois algorithm. I would recommend looking at Roots.jl here.

Reply:
Thanks for your comment, @ChrisRackauckas! The main reason we chose Newton's method is to obtain high-accuracy answers, and Newton-Raphson gives us that at reasonable speeds. Also, the method lends itself to other numeric types, which comes pretty handy when applying e.g. jet transport techniques. I'll surely take a look at Roots.jl, thanks!